src/cms/views/media/media_actions.py

```diff
@@ -1,38 +1,259 @@
 """
 This module contains view actions for media related objects.
 """
+import json
+
 from django.contrib.auth.decorators import login_required
-from django.shortcuts import redirect
+from django.db.models import Q
+from django.http import JsonResponse
+from django.shortcuts import get_object_or_404
+from django.core.exceptions import PermissionDenied
+from django.utils.translation import ugettext_lazy as _
 from django.views.decorators.http import require_POST
 
 from ...decorators import region_permission_required
-from ...models import Document, Region
+from ...models import Document, Region, Directory
+from ...utils.media_utils import attach_file
 
 
 @require_POST
 @login_required
 @region_permission_required
-# pylint: disable=unused-argument
-def delete_file(request, document_id, region_slug):
+def upload_file_ajax(request, region_slug=None):
+    """
+    View to create a file via an AJAX upload.
+
+    :param request: The current request
+    :type request: ~django.http.HttpResponse
+
+    :param region_slug: The slug of the current region
+    :type region_slug: str
+
+    :raises ~django.core.exceptions.PermissionDenied: If the user does not have permissions for this region.
+
+    :return: on success returns status 200
+    :rtype: ~django.http.JsonResponse
+    """
+    region = None
+    if region_slug is not None:
+        region = Region.objects.get(slug=region_slug)
+
+    directory_id = request.POST.get("parentDirectory", 0)
+    directory = None
+
+    if int(directory_id) != 0:
+        directory = Directory.objects.get(id=directory_id)
+        if directory.region != region:
+            raise PermissionDenied
+
+    if "upload" in request.FILES:
+        # TODO: Make sure only allowed files are uploaded by checking mime type here:
+        document = Document()
+        document.region = region
+        document.parent_directory = directory
+        document.name = request.FILES["upload"].name
+        attach_file(document, request.FILES["upload"])
+        document.save()
+        return JsonResponse({"success": True}, status=200)
```

Comment:
```suggestion
        return JsonResponse({"success": True})
```
The status 200 is the default one.